app/queries/invoices_query.rb

```diff
@@ -1,11 +1,12 @@
 # frozen_string_literal: true
 
 class InvoicesQuery < BaseQuery
-  def call(search_term:, status:, payment_status:, page:, limit:, filters: {}) # rubocop:disable Metrics/ParameterLists
+  def call(search_term:, status:, page:, limit:, filters: {}, customer_id: nil, payment_status: nil) # rubocop:disable Metrics/ParameterLists
     @search_term = search_term
 
-    invoices = base_scope.result.includes(:customer)
+    invoices = base_scope(customer_id:).result.includes(:customer)
```

Comment:
You could store the `customer_id` into an instance variable `@customer_id` and set it as `attr_reader`. This way you will not have to cascade it to the `search_params` method

Reply:
done, using `@customer_id` now